src/main.py

```
@@ -246,31 +288,95 @@
+                    write_timeout=TELEGRAM_WRITE_TIMEOUT,
+                    read_timeout=TELEGRAM_READ_TIMEOUT,
+                )
+        except TimedOut as e:
+            error("Telegram timeout while sending picture. %s", e)
+        except (NetworkError, TelegramError) as e:
+            await update.message.reply_text(f"Error sending picture: {str(e)}. Please try again later.")
+
+    elif isinstance(pic, list):
+        media_group = []  # Initilize empty list of media groups
+        for pic_file in pic:
+            media_group.append(InputMediaPhoto(open(pic_file, 'rb')))
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)